provider/verify.go

```diff
@@ -134,9 +133,7 @@ func (e VerifyHandler) LoginHandler(w http.ResponseWriter, r *http.Request) {
 // GET /login?site=site&user=name&address=someone@example.com
 func (e VerifyHandler) sendConfirmation(w http.ResponseWriter, r *http.Request) {
 
-	user, address := r.URL.Query().Get("user"), r.URL.Query().Get("address")
-	user = e.sanitize(user)
-	address = e.sanitize(address)
+	user, address, site := r.URL.Query().Get("user"), r.URL.Query().Get("address"), r.URL.Query().Get("site")
```

Comment:
this one is unclear to me either. The PR comment says "bluemonday as it is not necessary (any tags entered are encoded by the template library anyway, so there is no risk of injection)" but this method is a handler, and nothing prevents bad actors to pass anything in.

Reply:
the `sanitize`'d values are passed to 2 places:

- a handshake (which doesn't need them to be escaped or sanitised because it just handles them as a string)
- an email template builder (which does its own escaping)

Reply:
@paskal - can you take a look pls? To me, it looks like the `Sender` called after html/template was applied, and it can be safe to avoid sanitizing here. However, as far as I recall, this code was added to address some potential vulnerabilities, and I don't recall many details about it anymore. 

This sanitization was added by me on 06/03/22 "sanitize both user and address in verified provider". Do you recall if it was the result of some finding in remark42 around the same time?

Reply:
I did a little digging in to how these checks came about and found your comment @umputun here:

https://github.com/go-pkgz/auth/pull/119#issuecomment-1146404853

That explains why you were previously vulnerable to this attack, but are now double-encoding. The `text` -> `html` template change was the only required change in the end - it made the rest of the changes obsolete, and this PR effectively undoes those parts.